src/audio/copier/copier.c

```diff
@@ -457,6 +477,74 @@ static int create_dai(struct comp_dev *parent_dev, struct copier_data *cd,
 	return 0;
 }
 
+static int create_ipcgtw(struct comp_dev *parent_dev, struct copier_data *cd,
+			 struct comp_ipc_config *config,
+			 const struct ipc4_copier_module_cfg *copier)
+{
+#if CONFIG_COMP_IPCGTW
+	const struct comp_driver *drv;
+	const struct sof_uuid uuid = {0xa814a1ca, 0x0b83, 0x466c, {0x95, 0x87, 0x2f,
+				      0x35, 0xff, 0x8d, 0x12, 0xe8}};
+	struct comp_dev *dev;
+
+	drv = ipc4_get_drv((uint8_t *)&uuid);
+	if (!drv)
+		return -EINVAL;
+
+	/* create_endpoint_buffer() uses this value to choose between input and
+	 * output formats in copier config to setup buffer. For this purpose
+	 * IPC gateway should be handled similarly as host gateway.
+	 */
+	config->type = SOF_COMP_HOST;
+
+	int ret = create_endpoint_buffer(parent_dev, cd, config, copier, ipc4_gtw_none, false, 0);
+
+	if (ret < 0)
+		return ret;
+
+	dev = drv->ops.create(drv, config, &copier->gtw_cfg);
+	if (!dev) {
+		ret = -EINVAL;
+		goto e_buf;
+	}
+
+	list_init(&dev->bsource_list);
+	list_init(&dev->bsink_list);
+
+	if (cd->direction == SOF_IPC_STREAM_PLAYBACK) {
+		comp_buffer_connect(dev, config->core, cd->endpoint_buffer[cd->endpoint_num],
+				    PPL_CONN_DIR_COMP_TO_BUFFER);
+		cd->bsource_buffer = false;
+	} else {
+		comp_buffer_connect(dev, config->core, cd->endpoint_buffer[cd->endpoint_num],
+				    PPL_CONN_DIR_BUFFER_TO_COMP);
+		cd->bsource_buffer = true;
+	}
```

Comment:
between this code and the commit message, I am a bit lost of the problem statement.

In what cases would we send 'audio' over a SRAM window with IPC messages. We use DMAs for this.

Please clarify the usages that are improved by this solution.

Reply:
@serhiy-katsyuba-intel ping?

Reply:
IPC Gateway was used with some hardware like SueCreek there was no DMA. IPC Gateway has only dependency on IPC, so during bring-up of new hardware, once IPC is working one can start to run some tests/debugging with audio coming to/from host.